stl/src/vector_algorithms.cpp

```diff
@@ -3384,30 +3955,60 @@ namespace {
                     }
                 }
 
-                const int _Last_part_size_el = static_cast<int>(_Last_part_size / sizeof(_Ty));
-                const __m128i _Data1         = _mm_loadu_si128(reinterpret_cast<const __m128i*>(_Haystack));
-                _Test_whole_needle(_Data1, _Last_part_size_el);
+                if (_Last_part_size != 0) {
+                    const int _Last_part_size_el = static_cast<int>(_Last_part_size / sizeof(_Ty));
+                    __m128i _Data1;
+
+                    if (_Haystack_length_bytes >= 16) {
+                        _Data1 = _mm_loadu_si128(reinterpret_cast<const __m128i*>(_Haystack));
+                    } else {
+                        alignas(16) uint8_t _Tmp1[16];
+                        memcpy(_Tmp1, _Haystack, _Haystack_length_bytes);
+                        _Data1 = _mm_load_si128(reinterpret_cast<const __m128i*>(_Tmp1));
+                    }
+
+                    _Test_whole_needle(_Data1, _Last_part_size_el);
+                }
 
                 return static_cast<size_t>(_Found_pos);
             }
         }
 #endif // !_M_ARM64EC
-        const auto _Ptr_haystack = static_cast<const _Ty*>(_Haystack);
-        size_t _Pos              = _Haystack_length;
-        const auto _Needle_end   = static_cast<const _Ty*>(_Needle) + _Needle_length;
 
-        while (_Pos != 0) {
-            --_Pos;
+        template <class _Ty>
+        size_t _Dispatch_pos(
+            const void* const _First1, const size_t _Count1, const void* const _First2, const size_t _Count2) noexcept {
+            using namespace __std_find_meow_of_bitmap;
+
+#ifndef _M_ARM64EC
+            if (_Use_sse42()) {
+                const auto _Strat = _Pick_strategy<_Ty>(_Count1, _Count2, _Use_avx2());
 
-            for (auto _Ptr = static_cast<const _Ty*>(_Needle); _Ptr != _Needle_end; ++_Ptr) {
-                if (_Ptr_haystack[_Pos] == *_Ptr) {
-                    return _Pos;
+                if (_Strat == _Strategy::_Vector_bitmap) {
+                    if (_Can_fit_256_bits_sse(static_cast<const _Ty*>(_First2), _Count2)) {
+                        return _Impl_last_avx<_Ty>(_First1, _Count1, _First2, _Count2);
+                    }
+                } else if (_Strat == _Strategy::_Scalar_bitmap) {
+                    if (_Can_fit_256_bits_sse(static_cast<const _Ty*>(_First2), _Count2)) {
+                        alignas(32) _Scalar_table_t _Table = {};

```

Comment:
The aligning makes `memset` generated for `= {}` perform slightly faster and more predictable